### ISSUE TITLE
Bug: [no-unused-vars] unsupported Instantiation Expressions

### ISSUE THREAD

**Issue by nonsugarless** [typescript-eslint/typescript-eslint#5152]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.7.2&sourceType=module&code=GYVwdgxgLglg9mABAWwIYGsCmAhOAPAHgBUA+ACgDdUAbETALkSIEpEBvAKEUQCdMoQPJG0RVamRAF8A3B0kdMeAA5weURFACeSiQFkMOfMRKIAvBu2Y4wFAdyFSsoA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tidmUQAmtAG4BDaKgwBtSImjQO0SAF1wYAL4h1QA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### Repro Code

```typescript
function makeBox<T>(value: T) {
  return { value };
}
export type MakeBox<T> = typeof makeBox<T>;
```

### ESLint Config

```javascript
module.exports = {
  "rules": {
    "@typescript-eslint/no-unused-vars": ["error"]
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

No error since `T` is used

### Actual Result
```
'T' is defined but never used. 4:21 - 4:22
```
### Versions

| package | version |
| -- | -- |
| `@typescript-eslint/eslint-plugin` | `5.27.0` |
| `@typescript-eslint/parser` | `5.27.0` |
| `TypeScript` | `4.7.2` |
| `ESLint` | `8.15.0` |
| `node` | `web` |

**Comment by armano2** [comment]
this issue seem to be related to scope-manager, we are currently not visiting `typeParameters`
